Update README (#481)
jenkinsci/gradle-plugin: README.adoc

=====
Develocity `2024.1` or above is required starting from version `2.12` and above. See <<_short_lived_access_tokens>> for more infos.
=====
[IMPORTANT]
=====

> Starting from plugin version `2.13`, Maven extensions are not coming as bundled and require an explicit version configuration to be injected.
> When upgrading from a lower version, the injection will be turned off until the version is configured.

Reviewer: ```suggestion
Starting from plugin version `2.13`, the Maven extension is not bundled with the Jenkins Gradle plugin and requires an explicit version configuration to be injected.
When upgrading from an earlier version, the injection will be turned off until the version is configured.
```
